sklearn/model_selection/_split.py

```diff
@@ -1449,14 +1445,11 @@ class GroupShuffleSplit(ShuffleSplit):
     n_splits : int (default 5)
         Number of re-shuffling & splitting iterations.
 
-    test_size : float, int, None, optional
+    test_size : float, int, None, optional (default=0.2)
         If float, should be between 0.0 and 1.0 and represent the proportion
         of the dataset to include in the test split. If int, represents the
         absolute number of test groups. If None, the value is set to the
-        complement of the train size. By default, the value is set to 0.2.
-        The default will change in version 0.21. It will remain 0.2 only
-        if ``train_size`` is unspecified, otherwise it will complement
-        the specified ``train_size``.
+        complement of the train size.
```

Comment:
I'm not sure how you implement this by setting test_size=0.2 by default. We still need a placeholder value that acts like complement or 0.2 when unspecified. We still want to support the case that both train_size and test_size are set explicitly

Reply:
There's a validation function for `train_size` and `test_size`: `_validate_shuffle_split`.
It does what's described. I just added an error raise when both are None for `train_test_split`.

Reply:
I could be mistaken, but I don't think you're right. We agree on:

```py
ShuffleSplit() => (train=90%, test=10%)
ShuffleSplit(test_size=.2) => (train=80%, test=20%)
ShuffleSplit(test_size=.2, train_size=.5) => (train=50%, test=20%)
```

But while I think this PR (and IIRC the code before 0.19) does:

```py
ShuffleSplit(train_size=.8) => (train=80%, test=10%)
```

the documented (and desired) behaviour is:

```py
ShuffleSplit(train_size=.8) => (train=80%, test=20%)
```

Similarly

```py
ShuffleSplit(train_size=.99) => (train=99%, test=1%)
```
not an error.

Reply:
ahhh ok I misunderstood the desired behavior.

Reply:
I don't see how we can have a default `test_size=0.2` and both
```python
ShuffleSplit(train_size=.5) => (train=50%, test=50%)
ShuffleSplit(train_size=.5, test_size=.2) => (train=50%, test=20%)
```
there's no way to know that `test_size=0.2` has been explicitely set by a user.
What we could do is set the default to None and:
- set it to 0.2 if train_size is None
- complement train_size otherwise

Reply:
I moved this part to #13483 because it needs more work